packages/nodes-base/nodes/EmailReadImap/v2/EmailReadImapV2.node.ts

```diff
@@ -669,14 +669,19 @@ export class EmailReadImapV2 implements INodeType {
 		}
 
 		// When workflow and so node gets set to inactive close the connection
-		async function closeFunction() {
+		const closeFunction = async () => {
 			closeFunctionWasCalled = true;
 			if (reconnectionInterval) {
 				clearInterval(reconnectionInterval);
 			}
-			if (connection.closeBox) await connection.closeBox(false);
-			connection.end();
-		}
+			try {
+				if (connection.closeBox) await connection.closeBox(false);
+				connection.end();
```

Comment:
Should the `connection.end` be in a `finally` block?

Reply:
This cannot fail? https://github.com/mscdex/node-imap/blob/9918f08bc0bf5e3b0ed0e154fe98b3396e108606/lib/Connection.js#L306